File: baekjoon_all/10000+/boj_10707.py
# Solve 2023-03-07
# Update 2023-12-19

import sys

input = lambda : sys.stdin.readline().rstrip()

a, b, c, d, p = (int(input()) for _ in range(5))
x_cost = a * p
y_cost = b + max(0, p - c) * d
print(min(x_cost, y_cost))

File: baekjoon_all/10000+/boj_10707.cpp
// Solve 2022-06-30
// Update 2025-02-25

#include <bits/stdc++.h>

#define FASTIO ios_base::sync_with_stdio(false);cin.tie(NULL);
#define SIZE(v) (int)v.size()
#define ALL(v) v.begin(),v.end()
#define SETW(n, c) cout << setw(n) << setfill(c);
#define SETP(n) cout << fixed << setprecision(n);

using namespace std;
using ll = long long;
using uint = unsigned int;
using ull = unsigned long long;
using ld = long double;
using pii = pair<int, int>;

int main() {
    FASTIO;

    int a, b, c, d, p;
    cin >> a >> b >> c >> d >> p;

    int x_cost = a * p;
    int y_cost = b + max(0, p - c) * d;

    cout << min(x_cost, y_cost) << '\n';

    return 0;
}
